### PR TITLE
Rabbit Mutagen spawn and recipe change

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary
Content "Rabbit Mutagen Spawn and recipe change"

#### Purpose of change
Rabbit mutagen didn't spawn anywhere and was crafted only.  This will change that.  Also this is the first swing at differentiating mutagen ingredients and making them more specific to the mutagen being crafted.   

#### Describe the solution
Adds spawns for mutagen.  Adds a rabbit foot item, adds it as a dissection spawn.  Adds its to the recipe for rabbit mutagen.

If the ingredient changes are accepted then there will be a followup PR that adds specific ingredients and how to get them for each mutagen recipe.

#### Describe alternatives you've considered
None

#### Testing
TBD

### CODE REVIEW

**Comment by Maleclypse** [comment]
@Venera3 I wanted your opinion on this as I recall you being a key part of the discussions around creating specific ingredients that players need to hunt down to create mutagens.

**Comment by curstwist** [comment]
Can you add the leg to the refrigerated group in labs_mutagen.json

**Comment by Maleclypse** [comment]
> Can you add the leg to the refrigerated group in labs_mutagen.json

Is that different from 
```
  {
    "id": "lab_mutagen_ingredients_fridge",
    "type": "item_group",
    "subtype": "collection",
    "entries": [
      { "item": "meat", "prob": 40, "count": [ 1, 3 ] },
      { "item": "rabbit_foot", "prob": 40, "count": [ 1, 3 ] },
```

**Comment by jbytheway** [comment]
I like it.  I was imagining we would have something more generic like "<animal name> tissue sample".  On the one hand a rabbit's foot is a fun, more specific, alternative.  On the other hand, I think people will expect to be able to get it from regular butchery, not just dissection.

**Comment by Maleclypse** [comment]
> I like it. I was imagining we would have something more generic like " tissue sample". On the one hand a rabbit's foot is a fun, more specific, alternative. On the other hand, I think people will expect to be able to get it from regular butchery, not just dissection.

Yeah, so with the rabbits foot I was leaning into the idea that the blob isn't specifically using dna, at least not exclusively DNA but also what the creator is expecting the mutagen to do and why.  Kind of an emotional resonance.

**Comment by curstwist** [comment]
> 
> 
> > Can you add the leg to the refrigerated group in labs_mutagen.json
> 
> Is that different from
> 
> ```
>   {
>     "id": "lab_mutagen_ingredients_fridge",
>     "type": "item_group",
>     "subtype": "collection",
>     "entries": [
>       { "item": "meat", "prob": 40, "count": [ 1, 3 ] },
>       { "item": "rabbit_foot", "prob": 40, "count": [ 1, 3 ] },
> ```

nope, I'm just too sleepy to be reading PRs on mobile.  Thanks.

**Comment by Venera3** [comment]
I'm not sure I'd go with something as banal as their foot, there's a lot more to being a rabbit than what you can glean from a paw. Maybe rename it "rabbit sample" and describe it as "a collection of quintessential rabbit parts, that could explain what a rabbit was like without words"? 

Using something a bit more DNA-y, like bone marrow would make more sense in my head, but it would also introduce a *very* short rot timer to crafting mammal mutagen barring some weird preservation tool requirement. Down the line something like carrying a "mutagen starter" around and it imprinting on the creature as you dissect them would be the ideal solution, but it would obviously need a lot more code than this.

File: data/json/items/generic.json [hunk 1]
```diff
@@ -198,6 +198,21 @@
     "volume": "250 ml",
     "category": "spare_parts"
   },
+  {
+    "type": "GENERIC",
+    "id": "rabbit_foot",
+    "symbol": "q",
+    "color": "brown",
+    "name": { "str": "rabbit foot" },
```
Comment (Qrox) [inline data/json/items/generic.json:206]:
```suggestion
    "name": { "str": "rabbit foot", "str_pl": "rabbit feet" },
```

**Comment by Maleclypse** [comment]
> I'm not sure I'd go with something as banal as their foot, there's a lot more to being a rabbit than what you can glean from a paw. Maybe rename it "rabbit sample" and describe it as "a collection of quintessential rabbit parts, that could explain what a rabbit was like without words"?
> 
> Using something a bit more DNA-y, like bone marrow would make more sense in my head, but it would also introduce a _very_ short rot timer to crafting mammal mutagen barring some weird preservation tool requirement. Down the line something like carrying a "mutagen starter" around and it imprinting on the creature as you dissect them would be the ideal solution, but it would obviously need a lot more code than this.

I also wasn't sure how far we are going down the path of "the blob is not mutating DNA, it's making changes via interdimensional magic" and if that's the case we could include the idea of emotional resonance from creator expectations causing part of the direction a specific mutagen goes down.   Anyway I don't think that mutagen starter and old school thaumaturgy ideas are necessarily opposed when it comes to the blob.

File: data/json/items/generic.json [hunk 2]
```diff
@@ -198,6 +198,21 @@
     "volume": "250 ml",
     "category": "spare_parts"
   },
+  {
+    "type": "GENERIC",
+    "id": "rabbit_foot",
+    "symbol": "q",
+    "color": "brown",
+    "name": { "str": "rabbit foot", "str_pl": "rabbit feet" },
```
Comment (Maleclypse) [inline data/json/items/generic.json:206]:
```suggestion
    "name": { "str": "rabbit sample" },
```

**Review by actual-nh — APPROVED** [review]

**Pull request merged**